store/src/androidTest/kotlin/com/dropbox/external/store5/MarketMultithreadedAndroidTests.kt

```diff
@@ -0,0 +1,76 @@
+package com.dropbox.external.store5
+
+import com.dropbox.external.store5.fake.FakeDb
+import com.dropbox.external.store5.fake.FakeFactory
+import com.dropbox.external.store5.fake.FakeNotes
+import com.dropbox.external.store5.fake.OkTestMarket
+import com.dropbox.external.store5.fake.api.FakeApi
+import com.dropbox.external.store5.fake.model.Note
+import com.dropbox.external.store5.impl.MemoryLruCache
+import com.dropbox.external.store5.impl.SomeBroadcast
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import org.junit.Before
+import org.junit.Test
+import kotlin.concurrent.thread
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class MarketMultithreadedAndroidTests {
+    private val testScope = TestScope()
+    private lateinit var api: FakeApi
+    private lateinit var market: Market<String>
+    private lateinit var db: FakeDb
+    private lateinit var memoryLruCache: MemoryLruCache
+    private lateinit var factory: FakeFactory<String, Note, Note>
+
+    @Before
+    fun before() {
+        api = FakeApi()
+        market = OkTestMarket.build()
+        db = OkTestMarket.db
+        memoryLruCache = OkTestMarket.memoryLruCache
+        factory = FakeFactory(api)
+    }
+
+    @Test
+    fun write() = testScope.runTest {
+        val threads = mutableListOf<Thread>()
+        for (i in 0..10) {
+            threads.add(thread(start = true) {})
+        }
```

Comment:
I'm going to give this PR a more thorough read later, but I wanted to go ahead and throw in an observation here:

I strongly recommend avoiding using real threads in tests. [This blog post](https://publicobject.com/2022/03/14/uncertainty-in-tests/) gets into the reasons why.

Ofc the natural response is... "What do I do instead?" And the answer (at least, for things being run on CI) is to write single threaded tests that target specific bugs. Coroutines are a powerful tool for this, but it does require getting familiar with learning how to use `Channel` or `Deferred` to control concurrent flow of the coroutines.

Reply:
Thanks for taking a look and sharing link! Will rework 👍🏽

Reply:
Removed multithreaded tests. My plan is to return to this during migration of Store 4 tests (#449) and after bug bashing sample app (#446)